backend/sources/google/events-router/src/main/java/co/airy/core/sources/google/EventsRouter.java

```diff
@@ -75,51 +85,78 @@ private void startStream() {
                         return KeyValue.pair("skip", null);
                     }
 
-                    GoogleEventInfo googleEventInfo = GoogleInfoExtractor.extract(webhookEvent);
-                    googleEventInfo.setEventPayload(sourceEvent);
-                    googleEventInfo.setTimestamp(Instant.parse(webhookEvent.getSendTime()).toEpochMilli());
+                    final EventInfo eventInfo = InfoExtractor.extract(webhookEvent);
+                    eventInfo.setEvent(webhookEvent);
+                    eventInfo.setTimestamp(Instant.parse(webhookEvent.getSendTime()).toEpochMilli());
 
-                    if (!webhookEvent.isMessage()) {
-                        return KeyValue.pair(googleEventInfo.getAgentId(), null);
+                    if (!webhookEvent.isMessage() && !webhookEvent.hasContext()) {
```

Comment:
To discuss: Functionally we could let all records pass to the `flatMap`, but by filtering out anything that is either not a message or has no context we reduce the repartition store size. Is this worth it? cc @lucapette and @paulodiniz